server/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -50,10 +49,9 @@
  */
 public class TermsQueryBuilder extends AbstractQueryBuilder<TermsQueryBuilder> {
     public static final String NAME = "terms";
-    private static final TransportVersion VERSION_STORE_VALUES_AS_BYTES_REFERENCE = TransportVersions.V_7_12_0;
```

Comment:
Outdated, we can drop all code for pre-v7.12, in this case that means dropping the list backed `Values` and then collapsing `Values` into `BinaryValues`